gcsfs/core.py

```diff
@@ -957,17 +979,29 @@ def getxattr(self, path, attr):
         meta = self.info(path).get('metadata', {})
         return meta[attr]
 
-    def setxattrs(self, path, **kwargs):
-        """ Set user-defined metadata attributes
+    def setxattrs(self, path, content_type = None, content_encoding=None, **kwargs):
+        """ Set/delete/add writable metadata attributes
 
         Parameters
         ---------
-        kw_args : key-value pairs like field="value", where the values must be
-            strings. Replaces existing metadata.
+        content_type: str
+            Sets the Content-Type. Note: gcsfs defaults to 'application/octet-stream' on uploads
+        content_encoding: str
+            set 'gzip' + content_type 'text/plain' for transcoding (https://cloud.google.com/storage/docs/transcoding)
```

Comment:
I don't know what this means. What else might be accepted here?

Reply:
Suggested: "Specify 'gzip' as per https://cloud.google.com/storage/docs/transcoding guidance"

Not sure we would want cross-checking. Let the user read the document for proper combinations

Reply:
Is 'gzip' the only possible thing that can be given here, aside from None?

Reply:
`gzip` is the most commonly used and referenced as browsers can gunzip on-the-fly but specs lists 3: https://tools.ietf.org/html/rfc7231#section-3.1.2.1
Like for Content-Type, I would let the user decides the value to pass. No need for validation AFAIC